### PR TITLE
More fossilisation buttons

### PR DESCRIPTION
**Pull request by Oliveriver**
**Brief Description of What This PR Does**

* Adds a button to open the fossils folder in the museum
* Adds fossilise buttons to species details in the Auto-Evo exploring tool and Thriveopedia evolutionary tree
* Extracts SpeciesDetailsPanel to a new scene

Note the newly extracted SpeciesDetailsPanel doesn't include the heading anymore. I personally prefer it without the heading, but it could be replaced or conditionally set.

**Related Issues**

N/A

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/thriveopedia/pages/ThriveopediaMuseumPage.tscn
Comment (Kasterisk) [inline src/thriveopedia/pages/ThriveopediaMuseumPage.tscn:198]: The dialog box is too wide, I think lowering this to 300 would be good.

**Review by Kasterisk — CHANGES_REQUESTED** [review]

File: src/thriveopedia/pages/ThriveopediaMuseumPage.tscn
Comment (Oliveriver) [reply src/thriveopedia/pages/ThriveopediaMuseumPage.tscn]: Done. It does look better.

**Review by Kasterisk — APPROVED** [review]
With that one thing fixed, nothing seems out of ordinary.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (hhyyrylainen) [inline src/gui_common/SpeciesDetailsPanel.cs:34]: This should be rewritten to work when set even before this is attached to the scene tree.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (hhyyrylainen) [inline src/gui_common/SpeciesDetailsPanel.cs:65]: This print will happen when the species is null. Is it intended to allow setting the species to preview to null?

File: src/microbe_stage/MicrobeStage.tscn
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeStage.tscn:2221]: Accidental change here again.

File: src/auto-evo/AutoEvoExploringTool.cs
Comment (hhyyrylainen) [inline src/auto-evo/AutoEvoExploringTool.cs:716]: This check should be put in the PreviewSpecies property so that it skips applying the GUI control changes if the new value is the same as the old one.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
A few comments regarding how I think the code could be improved and I spotted an accidental change as well that should be reverted.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (Oliveriver) [reply src/gui_common/SpeciesDetailsPanel.cs:65]: This follows the existing pattern of `SpeciesPreview`.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (Oliveriver) [reply src/gui_common/SpeciesDetailsPanel.cs:39]: I've used a `ready` bool but I seem to remember you requesting a different approach to this in the past. I can't remember what that was. Let me know if you had a better approach in mind.

Comment (hhyyrylainen) [reply src/gui_common/SpeciesDetailsPanel.cs:39]: Make the control nullable, then check in the update method if it is null, and if it is then skip. And finally also add a call to that update method at the end of _Ready

**Review by hhyyrylainen — APPROVED** [review]
All my code related concerns have been addressed. I've not playtested.

**Comment by hhyyrylainen** [comment]
This is good to merge once the translation merge conflicts are resolved.

**Comment by Oliveriver** [comment]
Had some issues with submodule files being unexpectedly modified during the merge. The diff suggests I've put everything back to how it was, but you might want to double check when you merge this into master.

**Comment by hhyyrylainen** [comment]
As far as github shows anything, this PR doesn't change the git submodules.

**Comment by hhyyrylainen** [comment]
I noticed this PR removes the species details text entirely? What's the reason behind that. I see it used to be in the auto-evo exploring tool so @84634E1A607A might also want to take a look at this.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (84634E1A607A) [inline src/gui_common/SpeciesDetailsPanel.cs:63]: You could just check one variable. (And I think the 'ready' approach is used commonly elsewhere?)

File: src/gui_common/SpeciesDetailsPanel.tscn
Comment (84634E1A607A) [inline src/gui_common/SpeciesDetailsPanel.tscn:12]: I think this should be a `Control` so that it can be added in any Container. The current design will result in fixed margin and so on. It may create GUI designing issues.

File: src/auto-evo/AutoEvoExploringTool.tscn
Comment (84634E1A607A) [inline src/auto-evo/AutoEvoExploringTool.tscn:765]: Here you should preserve the PanelContainer, MarginContainer, VBoxContainer, MarginContainer and the Label.

File: src/auto-evo/AutoEvoExploringTool.tscn
Comment (84634E1A607A) [inline src/auto-evo/AutoEvoExploringTool.tscn:787]: This label.

**Review by 84634E1A607A — CHANGES_REQUESTED** [review]
Code logic is correct, but I have one GUI designing suggestion.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (hhyyrylainen) [reply src/gui_common/SpeciesDetailsPanel.cs:63]: I had the same thought but didn't ask for that as I think I've checked up to two variables like this... so checking 4 seems pretty minor, but I think just making one of the fields nullable would be a slight code improvement here.

Comment (Oliveriver) [reply src/gui_common/SpeciesDetailsPanel.cs:63]: Do you have a preference for which one field should be nullable then?

File: src/auto-evo/AutoEvoExploringTool.tscn
Comment (Oliveriver) [reply src/auto-evo/AutoEvoExploringTool.tscn:787]: I mentioned that label in the original post. I can leave it in if you would prefer, although I think it looks better in both places without it.

File: src/gui_common/SpeciesDetailsPanel.cs
Comment (hhyyrylainen) [reply src/gui_common/SpeciesDetailsPanel.cs:63]: I'd personally just pick the first of these fields.

File: src/auto-evo/AutoEvoExploringTool.tscn
Comment (84634E1A607A) [reply src/auto-evo/AutoEvoExploringTool.tscn:787]: I'm unsure about it, but I suppose it is fine then. However, I still want the SpeciesDetailsPanel to be a Control.

File: src/auto-evo/AutoEvoExploringTool.tscn
Comment (Oliveriver) [reply src/auto-evo/AutoEvoExploringTool.tscn:765]: The FossilisationButton exists outside of the second MarginContainer, so I can't move that out of the SpeciesDetailsPanel (the entire point of creating it was to have common fossilisation logic). Preserving the other three seems to prevent the FossilisationDialog showing, again for reasons I don't have time to figure out.

**Comment by 84634E1A607A** [comment]
I'll give it a try.

**Comment by hhyyrylainen** [comment]
I forgot I had already reviewed the code of this PR. So that means this is just waiting for @Oliveriver's opinion on #3982

**Review by hhyyrylainen — APPROVED** [review]
I think the new changes from the PR merged into this are probably fine, so just need to solve the merge conflicts and then this can be merged to master.

**Comment by hhyyrylainen** [comment]
I'll merge this once the tests pass.

**Pull request merged**
